### PR TITLE
Rename dev runtime to browser

### PR DESCRIPTION
**Pull request by timneutkens**
### Description

Builds on top of earlier changes, renames "dev" to "browser" to mirror earlier changes to reflect reality.

### CODE REVIEW

**Comment by timneutkens** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/vercel/turbo/8400?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#8400** <a href="https://app.graphite.dev/github/pr/vercel/turbo/8400?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#8399** <a href="https://app.graphite.dev/github/pr/vercel/turbo/8399?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8398** <a href="https://app.graphite.dev/github/pr/vercel/turbo/8398?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8397** <a href="https://app.graphite.dev/github/pr/vercel/turbo/8397?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8396** <a href="https://app.graphite.dev/github/pr/vercel/turbo/8396?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @timneutkens and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by arlyon — APPROVED** [review]

File: crates/turbopack-ecmascript-runtime/src/browser_runtime.rs [hunk 1]
```diff
@@ -35,12 +35,12 @@ pub async fn get_browser_runtime_code(
     let runtime_backend_code = embed_static_code(
         asset_context,
         match chunk_loading {
-            ChunkLoading::Edge => "dev/runtime/edge/runtime-backend-edge.ts".into(),
+            ChunkLoading::Edge => "browser/runtime/edge/runtime-backend-edge.ts".into(),
```
Comment (ForsakenHarmony) [inline crates/turbopack-ecmascript-runtime/src/browser_runtime.rs:38]: How can it be browser and edge at the same time?

File: crates/turbopack-ecmascript-runtime/js/package.json [hunk 2]
```diff
@@ -7,10 +7,10 @@
   "scripts": {
     "check": "run-p check:*",
     "check:nodejs": "tsc -p src/nodejs",
-    "check:dev-client": "tsc -p src/dev/client",
-    "check:dev-runtime-base": "tsc -p src/dev/runtime/base",
-    "check:dev-runtime-dom": "tsc -p src/dev/runtime/dom",
-    "check:dev-runtime-edge": "tsc -p src/dev/runtime/edge"
+    "check:browser-client": "tsc -p src/browser/client",
+    "check:browser-runtime-base": "tsc -p src/browser/runtime/base",
```
Comment (ForsakenHarmony) [inline crates/turbopack-ecmascript-runtime/js/package.json:11]: (commenting here to start a comment thread)

Maybe it should be `browser-dev` as the base runtime contains react refresh and hot module reloading code that's only used in dev?

File: crates/turbopack-ecmascript-runtime/src/browser_runtime.rs (hunk 1 above)
Comment (timneutkens) [reply crates/turbopack-ecmascript-runtime/src/browser_runtime.rs:38]: That's a refactor I'm still working on, previously `get_browser_runtime_code()` was called `get_dev_runtime_code` but it never used this for Node.js, instead it used `get_build_runtime_code`. Bit confusing naming so I've been working with @sokra to get this code into the right place.

File: crates/turbopack-ecmascript-runtime/js/package.json (hunk 2 above)
Comment (timneutkens) [reply crates/turbopack-ecmascript-runtime/js/package.json:11]: Yeah I'm going to split them up further in a follow-up PR. Wanted to keep the changes small as these broke tests a few times when I tried making larger changes to it.

Comment (ForsakenHarmony) [reply crates/turbopack-ecmascript-runtime/js/package.json:11]: I'd rather have the name follow functionality than the other way around (i.e. I'd prefer changing functionality first and then the name)

Comment (timneutkens) [reply crates/turbopack-ecmascript-runtime/js/package.json:11]: Already working on this change but it's actually more correct to not name it dev currently as it's used for builds as well at this point, that's what I'm actually working on, adding a production runtime

Comment (timneutkens) [reply crates/turbopack-ecmascript-runtime/js/package.json]: Pushed up the changes!

File: crates/turbopack-ecmascript-runtime/js/src/browser/dev/hmr-client/hmr-client.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-/// <reference path="../../shared/runtime-types.d.ts" />
+/// <reference path="../../../shared/runtime-types.d.ts" />
```
Comment (sokra) [inline crates/turbopack-ecmascript-runtime/js/src/browser/dev/hmr-client/hmr-client.ts:1]: That's weird

**Review by sokra — APPROVED** [review]

**Comment by timneutkens** [comment]
### Merge activity

* **Jun 19, 4:21 AM EDT**: @timneutkens started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/vercel/turbo/8400).

**Pull request merged**